### DISCUSSION TITLE
How to quickly add instance in cluster

### DISCUSSION THREAD

**Discussion by Doslin** [nats-io/k8s#729] [Q&A] [answered]
When I scale the number of pods in a statefulset, does that automatically scale the number of nodes in the nats cluster?

**Comment by caleblloyd** [comment, marked as answer]
Yes.

If you are using JetStream keep in mind that before scaling down you will need migrate all of the streams in the cluster off of the pods that are being scaled down.  For this reason, I would recommend vertical scaling for JetStream enabled clusters so that the number of pods stays the same.

If you are not using JetStream then horizontal scaling should work fine.
